src/mpid/ch3/channels/nemesis/netmod/mxm/mxm_send.c

```diff
@@ -299,16 +299,16 @@ int MPID_nem_mxm_send(MPIDI_VC_t * vc, const void *buf, MPI_Aint count, MPI_Data
             intptr_t last;
             MPI_Aint packsize = 0;
 
-            sreq->dev.segment_ptr = MPIR_Segment_alloc();
-            MPIR_ERR_CHKANDJUMP1((sreq->dev.segment_ptr == NULL), mpi_errno, MPI_ERR_OTHER,
-                                 "**nomem", "**nomem %s", "MPIR_Segment_alloc");
             MPIR_Pack_size_impl(count, datatype, &packsize);
 
             last = data_sz;
             if (packsize > 0) {
                 sreq->dev.tmpbuf = MPL_malloc((size_t) packsize, MPL_MEM_BUFFER);
                 MPIR_Assert(sreq->dev.tmpbuf);
-                MPIR_Segment_init(buf, count, datatype, sreq->dev.segment_ptr);
+
+                sreq->dev.segment_ptr = MPIR_Segment_alloc(buf, count, datatype);
+                MPIR_ERR_CHKANDJUMP1((sreq->dev.segment_ptr == NULL), mpi_errno, MPI_ERR_OTHER,
+                                     "**nomem", "**nomem %s", "MPIR_Segment_alloc");
                 MPIR_Segment_pack(sreq->dev.segment_ptr, 0, &last, sreq->dev.tmpbuf);
```

Comment:
The question might be unrelated to this commit: where do we free the segment ?

Reply:
Fixed.